### PR TITLE
Project widget V1

### PR DESCRIPTION
**Pull request by caleybrock**
Recent projects and static new project list.

screen shot 2017-07-06 at 9 05 00 am

Moved ContentContainer to make it more generalized.

### CODE REVIEW

**Comment by caleybrock** [comment]
I added this to a page by adding the following:

`#projects-widget` to a haml file.

To a js file:

```
  $.ajax({
    method: 'GET',
    url: `/v3/channels`,
    dataType: 'json'
  }).done(projectLists => {
    const widget = document.getElementById('projects-widget');
    projectLists.sort((a, b) => {
      if (a.updatedAt < b.updatedAt) {
        return 1;
      } else {
        return -1;
      }
    });
    projectLists = projectLists.filter(project => {
      return !project.hidden;
    });
    projectLists = projectLists.slice(0,4);
    let convertedProjectList = [];
    for (let i = 0; i < projectLists.length; i++){
      let convertedProject = {
        name: projectLists[i].name,
        channel: projectLists[i].id,
        thumbnailUrl: projectLists[i].thumbnailUrl,
        type: projectLists[i].projectType,
        updatedAt: projectLists[i].updatedAt
      };
      convertedProjectList.push(convertedProject);
    }
    ReactDOM.render(
      <Provider store={getStore()}>
        <ProjectWidget projectList={convertedProjectList}/>
      </Provider>,
      widget);
  });
});
```

^This is not a super clean way to pass data in, just a way to prove we have the data setup in a way that will work with our components.

This work item didn't include adding this to pages yet, but the student page and course overview pages might be ones to come.

Component is 970px wide, our new width going forward, so it should work well on other pages, but was tested on the projects page.

File: apps/src/templates/projects/NewProjectButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+      </div>
+    );
+  },
+
+  render() {
+    return (
+      <div style={styles.fullsize}>
+        <div style={styles.description}>{i18n.projectStartNew()}</div>
+        {this.renderButtons()}
```
Comment (Bjvanminnen) [inline apps/src/templates/projects/NewProjectButtons.jsx]: I would prefer just inlining renderButtons here

File: apps/src/templates/projects/PersonalRecentProjects.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        }
+      </div>
+    );
+  },
+
+  render() {
+    return (
+      <div style={styles.grid}>
+        <div style={styles.description}>{i18n.projectsContinueWorking()}</div>
+        {this.renderProjectCardList(this.props.projectList)}
```
Comment (Bjvanminnen) [inline apps/src/templates/projects/PersonalRecentProjects.jsx]: I would again recommend inlining renderProjectCardList.

File: apps/src/templates/projects/ProjectCard.jsx [hunk 3]
```diff
@@ -248,11 +249,20 @@ const ProjectCard = React.createClass({
...
 
-          <div style={styles.lastEdit}>
-            {this.renderArrowIcon()}
-            {i18n.published()}:&nbsp;
-            <time style={styles.bold} className="versionTimestamp" dateTime={projectData.publishedAt}> {this.getLastModifiedTimestamp(projectData.publishedAt)}</time>
-          </div>
+          {(this.props.currentGallery !== 'personal' && projectData.publishedAt) &&
+            <div style={styles.lastEdit}>
+              {i18n.published()}:&nbsp;
+              <time style={styles.bold} className="versionTimestamp" dateTime={projectData.publishedAt}> {this.getLastModifiedTimestamp(projectData.publishedAt)}</time>
```
Comment (Bjvanminnen) [inline apps/src/templates/projects/ProjectCard.jsx]: nit: would prefer to break up these long lines (here and 263)

File: apps/src/templates/projects/ProjectWidget.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        heading={i18n.projects()}
+        linkText={i18n.projectsViewProjectGallery()}
+        link="/projects"
+        showLink={true}
+        isRtl={false}
+      >
+        {this.props.projectList.length > 0 &&
+          <PersonalRecentProjects projectList={this.props.projectList} />
+        }
+        <NewProjectButtons />
```
Comment (Erin007) [inline apps/src/templates/projects/ProjectWidget.jsx]: nit: space before />

File: apps/src/templates/projects/PersonalRecentProjects.jsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+import {personalProjectDataPropType} from './projectConstants';
+import color from "../../util/color";
+import i18n from "@cdo/locale";
+import Radium from 'radium';
+
+const styles = {
+  grid: {
+    width: '100%'
+  },
+  link: {
```
Comment (Erin007) [inline apps/src/templates/projects/PersonalRecentProjects.jsx]: Should we use the styling for the linkToViewAll in ContentContatiner here, so it's consistent?

File: apps/src/templates/projects/ProjectCard.story.jsx [hunk 6]
```diff
@@ -12,33 +12,58 @@ const defaultData = {
...
+};
+
+const publishedToPublicPersonal = {
+  ...personalData,
   publishedToPublic: true
 };
 
-const PUBLISHED_CLASS = {
-  ...defaultData,
+const publishedToClassPersonal = {
```
Comment (Erin007) [inline apps/src/templates/projects/ProjectCard.story.jsx:35]: I'm confused about why some of these const are CAPS_SNAKE and some are lowerCamel. Should they be consistent?

Comment (caleybrock) [reply apps/src/templates/projects/ProjectCard.story.jsx:35]: I was also confused. Updating to all lowerCamel.

File: apps/src/templates/projects/PersonalRecentProjects.jsx (hunk 5 above)
Comment (caleybrock) [reply apps/src/templates/projects/PersonalRecentProjects.jsx]: This was unused CSS, removed.

**Comment by caleybrock** [comment]
Data exists but will probably be different because not all pages are set up the same. I'll talk to poorva/tanya about if we want the V1 version on any pages, before implementing V2.

**Comment by caleybrock** [comment]
@Bjvanminnen updated based on review.

File: apps/src/templates/projects/ProjectCard.jsx [hunk 7]
```diff
@@ -163,11 +164,11 @@ const ProjectCard = React.createClass({
...
 
   renderArrowIcon() {
-  // Only the student can access the menu options when viewing their personal projects.
-    if (this.props.currentGallery === 'personal') {
-     let icon = this.state.actionsOpen ? 'chevron-up' : 'chevron-down';
-     return (
-       <FontAwesome icon={icon} style={styles.arrowIcon} onClick={this.toggleActionBox}/>
+    if (!this.props.hideActions) {
+      // Only the student can access the menu options when viewing their personal projects.
+      let icon = this.state.actionsOpen ? 'chevron-up' : 'chevron-down';
```
Comment (Bjvanminnen) [inline apps/src/templates/projects/ProjectCard.jsx]: nit: could be const

**Pull request merged**